### ISSUE TITLE
gall: return of the 2nd watch-ack

### ISSUE THREAD

**Issue by liam-fitzgerald** [urbit/urbit#4687]
**Describe the bug**
@tacryt-socryp told me today that the 2nd watch-ack bug is supposed to be fixed. However, a fresh reset of the ropsten testnet reveals this not to be the case. Upon processing the pre-release/next-userspace OTA, ships are spewing 2nd watch-acks for a significant majority of the resources that they're pulling. The ropsten testnet had a network continuity breach maybe an a hour or two before I deployed this, so there's little time for the subscriptions to get out of sync. Is there anything that the userspace code could be doing to aggravate this?
<img width="1368" alt="Screen Shot 2021-03-30 at 9 35 40 am" src="https://user-images.githubusercontent.com/46801558/112912944-14ebe000-913c-11eb-9f96-1410de5ff6dd.png">
 

**To Reproduce**
Is a |verb trace worth it here? Happy to step through the deploy, if needed

**Notify maintainers**
cc: @belisarius222 @joemfb @philipcmonk

**Comment by belisarius222** [comment]
The Gall request queuing has not been completely fixed. A full fix will require writing a new state machine in Gall to properly align incoming acks with their outgoing requests.

Not sure offhand what would trigger this more often, but it's a known bug.

—
~rovnys-ricfer
https://urbit.org

**Comment by liam-fitzgerald** [comment]
I thought that i was supposed to be fixed for subscriptions that were not already affected?

**Comment by belisarius222** [comment]
No, the queue can still become desynchronized. See https://github.com/urbit/urbit/pull/4454

—
~rovnys-ricfer
https://urbit.org

**Comment by philipcmonk** [comment]
As mentioned to @tacryt-socryp, we should add this printf in +mo-handle-sys-way.  If it's the issue we're aware of where we need nonces to specify the distinct subscription lifetime, this should not fire.  If it's a different bug, this would fire immediately before the 2nd watch ack printf.

```
          =?  stand  ?=(~ stand)
            ~&  [%missing wire hen]
            (~(put to *(qeu remote-request)) %missing)
```
